File: python/general/1207. 独一无二的出现次数.py
from typing import List
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        d = dict()
        for i in arr:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
        s = set(d.values())
        return len(s) == len(d)

File: go/general/1207. 独一无二的出现次数.go
package main

func uniqueOccurrences(arr []int) bool {
	dict := make(map[int]int)
	for _, v := range arr {
		dict[v] += 1
	}
	set := make(map[int]interface{})
	for _, v := range dict {
		set[v] = struct{}{}
	}
	return len(set) == len(dict)
}
